drush.complete.sh

```diff
@@ -24,9 +24,41 @@ __drush_ps1() {
     __DRUPAL_SITE="$DRUPAL_SITE"
   fi
 
+  # Set DRUSH_PS1_SHOWCOLORHINTS to a non-empty value to get color hints in your
+  # Drush PS1 prompt: red for a production environment, green for local, and
+  # blue for anything else.
+  if [ -n "${__DRUPAL_SITE-}" ] && [ -n "${DRUSH_PS1_SHOWCOLORHINTS-}" ]; then
+    __drush_ps1_colorize_alias
+  fi
+
   [[ -n "$__DRUPAL_SITE" ]] && printf "${1:- (%s)}" "$__DRUPAL_SITE"
 }
 
+__drush_ps1_colorize_alias() {
+  if [[ -n ${ZSH_VERSION-} ]]; then
+    local COLOR_RED='%F{red}'
+    local COLOR_GREEN='%F{green}'
+    local COLOR_BLUE='%F{blue}'
+    local COLOR_CLEAR='%f'
+  else
+    # Using \[ and \] around colors is necessary to prevent issues with command
+    # line editing/browsing/completion.
```

Comment:
I took these color codes from [git/git-prompt.sh](https://github.com/git/git/blob/master/contrib/completion/git-prompt.sh), and I'm taking their word for it on this comment.